blockchain/blockchain.go

```diff
@@ -765,15 +770,47 @@ func revertHead(txn db.Transaction) error {
 	if err = txn.Delete(db.BlockHeaderNumbersByHash.Key(header.Hash.Marshal())); err != nil {
 		return err
 	}
+	if blockNumber != 0 { // Don't send block before genesis
```

Comment:
do you think introducing new variable `genesisBlock := blockNumber == 0`
and using it later like:
```
if !genesisBlock {
....
}
...
if genesisBlock {
...
}
```
would improve readability or it's unnecessary?

Reply:
I like it.